tensorly/regression/cp_plsr.py

```diff
@@ -178,14 +188,29 @@ def predict(self, X):
                 f"Training X has shape {self.X_shape_}, while the new X has shape {T.shape(X)}"
             )
         X -= self.X_mean_
-        factors_kr = khatri_rao(self.X_factors, skip_matrix=0)
-        unfolded = unfold(X, 0)
-        scores = T.lstsq(factors_kr, T.transpose(unfolded))[0]  # = Tnew
-        estimators = T.lstsq(self.X_factors[0], self.Y_factors[0])[0]
-        return (
-            T.dot(
-                T.dot(T.transpose(scores), estimators), T.transpose(self.Y_factors[1])
+        X_projection = T.zeros((T.shape(X)[0], self.n_components), **T.context(X))
+        for component in range(self.n_components):
+            X_projection = T.index_update(
```

Comment:
Does this mean that the Bro paper algorithm did in fact work in the end? I'm not seeing you subtracting from X_orig like we had previously discussed...

Reply:
Yes. The subtracting from the original Y turned out to be not a problem at all given the 0 columns elsewhere. Every component explains exactly what we have subtracted each round, in this case, the project of X to Y. Therefore, the definition of $R^2$ is rightfully different for X and Y; they are not symmetrical (surprise, surprise).